### PR TITLE
Test Updates For TF 2.4

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Added a `tf2_4.yml` buildspec to test the changes on a custom version of tf 2.4.0 gpu
- TF 2.4 returns 16 tenors for the BIAS collection  
- Expanded the `is_profiler_supported` function to include tf 2.4.0

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/tensorflow2/test_estimator.py [hunk 1]
```diff
@@ -71,10 +71,11 @@ def test_linear_classifier(out_dir, tf_eager_mode, saveall):
         # vanilla TF 2.2: all = 214, loss = 2, weights = 1, gradients = 0, biases = 12, optimizer variables = 0, metrics = 0, others = 199
         # AWS-TF 2.2: all = 219, loss = 2, weights = 1, gradients = 2, biases = 12, optimizer variables = 5, metrics = 0, others = 197
         # AWS-TF 2.1: all = 226, loss = 2, weights = 1, gradients = 2, biases = 12, optimizer variables = 5, metrics = 0, others = 204
+        # AWS-TF 2.4: all = 229, loss = 2, weights = 1, gradients = 2, biases = 16, optimizer variables = 5, metrics = 0, others = 197
         assert len(tnames) >= 2 + 1 + 12
         assert len(trial.tensor_names(collection=CollectionKeys.LOSSES)) == 2
         assert len(trial.tensor_names(collection=CollectionKeys.WEIGHTS)) == 1
-        assert len(trial.tensor_names(collection=CollectionKeys.BIASES)) == 12
+        assert len(trial.tensor_names(collection=CollectionKeys.BIASES)) >= 12
```
Comment (ndodda-amazon) [inline tests/tensorflow2/test_estimator.py:78]: If we know the exact number of tensors for TF 2.4, can we do a conditional assertion?

```
assert len == 16 if is_tf_2_4 else assert len == 12
```

**Review by ndodda-amazon — CHANGES_REQUESTED** [review]
Looks good, left a minor comment.

File: tests/tensorflow2/test_estimator.py (hunk 1 above)
Comment (NihalHarish) [reply tests/tensorflow2/test_estimator.py:78]: Yeah, better.

File: smdebug/tensorflow/utils.py [hunk 2]
```diff
@@ -428,4 +428,4 @@ def is_tf_version_greater_than_2_4_x():
 
 
 def is_profiler_supported_for_tf_version():
-    return is_tf_version_2_2_x() or is_tf_version_2_3_x()
+    return is_tf_version_2_2_x() or is_tf_version_2_3_x() or is_tf_version_greater_than_2_4_x()
```
Comment (ndodda-amazon) [inline smdebug/tensorflow/utils.py:431]: Two nits:
- Maybe we can just simplify this and do `return version.parse("2.2.0") <= TF_VERSION`
- Naming schema clashes with `is_tf_version_greater_than_2_4_x`, we should just rename this function to `is_tf_version_greater_than_2_2_x` or rename `is_tf_version_greater_than_2_4_x` appropriately.

**Review by ndodda-amazon — APPROVED** [review]

**Pull request merged**